File: 0257-binary-tree-paths/0257-binary-tree-paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    vector<string> ans;

    void dfs(TreeNode* r, string s)
    {
        if (!r) return;
        s += to_string(r -> val);        
        if (!r -> left && !r -> right)
        {
            ans.push_back(s);
            return;
        }        
        s += "->";
        dfs(r -> left, s);
        dfs(r -> right, s);
    }
    
public:
    vector<string> binaryTreePaths(TreeNode* root) {
        string s = "";
        dfs(root, s);
        return ans;
    }
};

File: 0257-binary-tree-paths/0257-binary-tree-paths.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {string[]}
 */
var binaryTreePaths = function(root) {
    let ans = [];
    const dfs = (r, s) => 
    {
        if (!root) return;
        if (r.left === null && r.right === null) 
        {
            ans.push(s + r.val);
            return;
        }
        
        if (r.left) dfs(r.left, s + r.val + "->");
        if (r.right) dfs(r.right, s + r.val + "->");
    };
    
    dfs(root, "");
    return ans;
}